### PR TITLE
feat(marmot-app): record an epoch_stall_backfill_armed forensic audit row at the epoch-stall arm decision

### PR DESCRIPTION
**Pull request by jgmontoya**
---

<a href="https://stagereview.app/marmot-protocol/mdk/pull/919">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * Added a new forensic audit log event for when epoch-stall backfill is armed, including the stalled epoch and backfill threshold.
  * Extended the audit log schema and JSON serialization to support the new `epoch_stall_backfill_armed` event type.
* **Bug Fixes**
  * Improved forensic upload scheduling to reflect whether backfill was armed before replay, even if replay fails.
  * Avoided duplicate audit entries for repeated triggers at the same stalled epoch.
* **Tests**
  * Added end-to-end tests covering threshold behavior, schema/event round-tripping, and upload generation without visible activity.

### CODE REVIEW

**Comment by ghost** [comment]
Ready to review this PR? Stage has broken it down into 5 individual chapters for you:

| | Title |
|---|-------|
| 1 | [Define epoch_stall_backfill_armed audit event schema](https://stagereview.app/marmot-protocol/mdk/pull/919/chapters/1) |
| 2 | [Expose stall threshold and recording logic](https://stagereview.app/marmot-protocol/mdk/pull/919/chapters/2) |
| 3 | [Trigger audit recording during sync](https://stagereview.app/marmot-protocol/mdk/pull/919/chapters/3) |
| 4 | [Wire backfill reporting into account worker](https://stagereview.app/marmot-protocol/mdk/pull/919/chapters/4) |
| 5 | [Test backfill audit logging and uploads](https://stagereview.app/marmot-protocol/mdk/pull/919/chapters/5) |

<a href="https://stagereview.app/marmot-protocol/mdk/pull/919">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://stagereview.app/assets/gh-open-in-stage-dark.svg">
    Open in Stage
  </picture>
</a>

<sub>Chapters generated by <a href="https://stagereview.app">Stage</a> for commit db422d59df4acaf661111f7ceb657bc6a9504c3f on Jul 20, 2026 5:41pm UTC.</sub>

File: crates/marmot-app/src/client/sync.rs [hunk 1]
```diff
@@ -337,10 +337,16 @@ impl AppClient {
...
+        if self.epoch_stall.observe_undecryptable(
+            group_id.clone(),
+            message_id_hex.to_owned(),
+            record.epoch,
+        ) {
+            // Record the arm decision before the replay side effect runs (the
+            // worker seam calls run_pending_epoch_backfill after this returns).
+            // Best-effort, fire-and-forget: recording can never block or fail
+            // the backfill.
+            self.record_epoch_stall_backfill_armed(&group_id, record.epoch.0);
```
Comment (erskingardner) [inline crates/marmot-app/src/client/sync.rs:349]: **[P2] Ensure this arm-only row triggers a tracker upload**

The threshold-crossing delivery intentionally makes `next_event` return with an empty `SyncSummary` (the existing `next_event_backfill` test pins that behavior). The account worker then runs the pending backfill, but only schedules `schedule_audit_log_tracker_update` when the summary contains a joined group, message, or event. Therefore this new `epoch_stall_backfill_armed` row stays local unless unrelated visible activity happens later; if the replay fails or produces no visible follow-up, the configured field tracker misses the exact diagnostic this change is adding.

Please thread whether a backfill was armed/run into tracker scheduling for the startup, receive, and catch-up paths (for example, have `run_pending_epoch_backfill` return whether it did work), and add a runtime-level tracker test that verifies an arm-only empty-summary path uploads the new row.

**Review by erskingardner — COMMENTED** [review]
One material integration gap found in the epoch-stall audit path.

File: crates/marmot-app/src/client/sync.rs (hunk 1 above)
Comment (jgmontoya) [reply crates/marmot-app/src/client/sync.rs:349]: Good catch, fixed!

**Review by erskingardner — APPROVED** [review]
